spec/models/email_unpublishing_processor_spec.rb

```diff
@@ -0,0 +1,234 @@
+      "schema_name" => "redirect",
+      "base_path" => "/foo",
+      "locale" => "en",
+      "publishing_app" => "whitehall",
+      "redirects" => [
+        {
+          "path" => "/government/publications/foo/html-attachment-foo",
+          "type" => "exact",
+          "destination" => "/government/publications/foobar",
+        },
+      ],
+      "public_updated_at" => public_updated_at,
+      "content_id" => content_id,
+      "govuk_request_id" => govuk_request_id,
+      "payload_version" => 199,
+    }
+  end
+
+  def unpublish_alert_was_called
+    expect(mock_unpublishing_alert).to have_received(:trigger)
+  end
+
+  def unpublish_alert_was_not_called
+    expect(mock_unpublishing_alert).not_to have_received(:trigger)
+  end
+
+  before :each do
+    allow(UnpublishingAlert).to receive(:new).and_return(mock_unpublishing_alert)
+  end
+
+  describe "#process" do
+    describe "in an unknown unpublishing scenario" do
+      let(:document) do
+        {
+          "document_type" => "woops",
+          "schema_name" => "woops",
+          "base_path" => "/foo",
+          "public_updated_at" => public_updated_at,
+          "content_id" => content_id,
+        }
+      end
+
+      it "does not trigger an email-api client call and logs a message" do
+        processor.process(message)
+
+        unpublish_alert_was_not_called
+        expect(logger).to have_received(:info).with(
+          "not triggering bulk unsubscription and alert for document with a unknown unpublishing scenario. Document type: #{document['document_type']}",
+        )
+      end
+    end
+
+    describe "following published_in_error_with_url event" do
+      let(:unpublishing_scenario) { :published_in_error_with_url }
+      let(:alternative_path) { "/foo/alternative" }
+      let(:document) { published_in_error_payload }
+
+      it "acknowledges and triggers an email-api client call for a correctly tagged document" do
+        processor.process(message)
+
+        unpublish_alert_was_called
+        message_acknowledged
+      end
+
+      context "a document without a base_path" do
+        before { document.delete("base_path") }
+
+        it "acknowledges but doesn't trigger the email" do
+          processor.process(message)
+
+          unpublish_alert_was_not_called
+          message_acknowledged
+        end
+      end
+
+      context "a document without a public_updated_at" do
+        before { document.delete("public_updated_at") }
+
+        it "acknowledges but doesn't trigger the email" do
+          processor.process(message)
+
+          unpublish_alert_was_not_called
+          message_acknowledged
+        end
+      end
+
+      context "a document without a content_id" do
+        before { document.delete("content_id") }
+
+        it "acknowledges but doesn't trigger the email" do
+          processor.process(message)
+
+          unpublish_alert_was_not_called
+          message_acknowledged
+        end
+      end
```

Comment:
Since these are the same for all the cases, move them to a `shared_examples` block.

Reply:
I think in the change notifiaction the format was:

```
# Title

## Body:
Title
```

optionally as a link for both.
It's a bit odd but I assume the Body section is still needed, no?